aiogram/types/message.py

```diff
@@ -1714,6 +1714,26 @@ def copy_to(
             reply_markup=reply_markup,
         )
 
+    def get_url(self, force_private: bool = False) -> str:
+        """
+        Returns message URL. Cannot be used in private (one-to-one) chats.
+        If chat has a username, returns URL like https://t.me/username/message_id
+        Otherwise (or if {force_private} flag is set), returns https://t.me/c/shifted_chat_id/message_id
+
+        :param force_private: if set, a private URL is returned even for a public chat
+        :return: string with full message URL
+        """
+        if self.chat.type == "private":
+            raise TypeError("Invalid chat type!")
```

Comment:
I suggest to return `None` instead of raising an errror.

Usage with error:
```python
try:
    url = message.get_url()
except TypeError:
    pass
else:
    # action
```

Usage with None:
```python
url = message.get_url()
if url is not None:
    # action
```

Reply:
My approach is based on the current implementation from 2.x
(https://github.com/aiogram/aiogram/blob/dev-2.x/aiogram/types/message.py#L277)

I guess we should discuss this in group chat

Reply:
A note to myself: `if self.chat.type == "private"` should be `self.chat.type in ("private", "group")`